kernel/src/process.rs

```diff
@@ -1142,14 +1144,24 @@ impl<C: Chip> ProcessType for Process<'a, C> {
     unsafe fn free(&self, _: *mut u8) {}
 
     #[allow(clippy::cast_ptr_alignment)]
```

Comment:
Is this clippy warning still valid?

Reply:
Looks like it:
```
$ ../tools/run_clippy.sh
    Checking tock-registers v0.5.0 (/Volumes/code/helena-project/tock/libraries/tock-register-interface)
    Checking tock-cells v0.1.0 (/Volumes/code/helena-project/tock/libraries/tock-cells)
    Checking kernel v0.1.0 (/Volumes/code/helena-project/tock/kernel)
error: casting from `*const u8` to a more-strictly-aligned pointer (`*mut *mut u8`) (1 < 8 bytes)
    --> src/process.rs:1160:17
     |
1160 |                 (self.mem_end() as *mut *mut u8).offset(-(grant_num + 1));
     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: `#[deny(clippy::cast_ptr_alignment)]` on by default
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cast_ptr_alignment

error: aborting due to previous error

error: could not compile `kernel`.
```

Reply:
This is beyond the scope of this pull-request, but this seems like a valid warning from Clippy. How are we making sure that `self.mem_end` is indeed aligned on a word?

Reply:
https://github.com/tock/tock/issues/1739 -- it's defined by MPU region granularity, which isn't smaller than a word on any of our platforms, but we should explain that a bit.  Opened issue to capture that  later.